scripts/checks/check_disease_synonyms.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import re
+import argparse
+import MySQLdb
+import django
+import configparser
+from difflib import SequenceMatcher
+from sentence_transformers import SentenceTransformer
+from sklearn.metrics.pairwise import cosine_similarity
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'gene2phenotype_app')
+django.setup()
+from gene2phenotype_app.utils import clean_string
+
+"""
+Description: Script to compare the similarity between the G2P diseases names and associated synonyms.
+             Writes to the output file the diseases that have similarity score below the cutoff value.
+
+Options
+        --config : Config file with connection details to the G2P database (mandatory)
+                File format is the following:
+                        [g2p_database]
+                        host = <>
+                        port = <>
+                        user = <>
+                        password = <>
+                        name = <>
+        --cutoff : Score value to filter the results
+        --output_file : Name of the output file
+
+Requirements: Add the gene2phenotype_app project path to PYTHONPATH (export PYTHONPATH=_path_to_gene2phenotype_)
+"""
+
+
+def dump_g2p_diseases(db_host: str, db_port: int, db_name: str, user: str, password: str) -> dict[str, list]:
+    """
+    Queries the G2P database to dump all the diseases that have a disease synonym.
+
+    Args:
+        db_host (str): G2P database host name
+        db_port (int): G2P database port number
+        db_name (str): G2P database name
+        user (str): user with read access
+        password (str): password
+
+    Returns:
+        dict[str, list]: All disease names with associated synonyms
+    """
+    diseases = {}
+
+    db = MySQLdb.connect(host=db_host, port=db_port, user=user, passwd=password, db=db_name)
+    cursor = db.cursor()
+
+    sql = """   SELECT d.name, s.synonym
+                FROM disease d
+                LEFT JOIN disease_synonym s ON s.disease_id = d.id
+                WHERE s.synonym IS NOT NULL
+          """
+
+    cursor.execute(sql)
+    data = cursor.fetchall()
+    for row in data:
+        disease_name = row[0]
+        disease_synonym = row[1]
```

Comment:
Here again with my defaultdict suggestion 🫠 , it just makes it faster here 
```
from collections import defaultdict
diseases = defaultdict(list)
for row in data:
    disease_name = row[0]
    disease_synonym = row[1]
    diseases[disease_name].append(disease_synonym)
```

this would allow us not to need to an if else check

Reply:
Updated